packages/sdks/snippets/react/src/routes/blueprints/ProductDetails.tsx

```diff
@@ -0,0 +1,47 @@
+import { BuilderContent, fetchOneEntry } from '@builder.io/sdk-react';
+import { useEffect, useState } from 'react';
+
+export default function ProductDetails() {
+  const [productDetails, setProductDetails] = useState<BuilderContent | null>(
+    null
+  );
+
+  useEffect(() => {
+    let isMounted = true;
+    const fetchProductDetails = async () => {
+      const productData = await fetchOneEntry({
+        model: 'product-details',
+        apiKey: 'ee9f13b4981e489a9a1209887695ef2b',
+        query: {
+          'data.handle': 'jacket',
+        },
+      });
+      if (isMounted) {
+        setProductDetails(productData);
+      }
+    };
+    fetchProductDetails();
+
+    return () => {
+      isMounted = false;
+    };
+  }, []);
```

Comment:
I don't see the point of `isMounted`. What does it achieve that can't be done without it, like so:

```suggestion
  useEffect(() => {
    fetchOneEntry({
        model: 'product-details',
        apiKey: 'ee9f13b4981e489a9a1209887695ef2b',
        query: {
          'data.handle': 'jacket',
        },
      }).then(productData => {
        setProductDetails(productData);
      }
  }, []);
```

Reply:
@samijaber I was reading about abort controllers, and then the react doc suggested the below:
<img width="941" alt="Screenshot 2025-03-05 at 10 28 27 AM" src="https://github.com/user-attachments/assets/8139b596-26e7-4514-a991-125187b8e5fe" />

I'm totally with you on simplifying the code, and I had the same code as yours before pushing it to the PR. The only reason I added a flag is to stop fetching twice and updating the state after unmounts.

Reply:
I'd ignore this. since your `useEffect` has an empty dependencies array, it will only run once no matter what, so there's no point in aborting the fetch or ignoring the result.

The advice you screenshotted is relevant when `useEffect` has items in its dependencies array, which means it will execute more than once.

Reply:
As for async/await: it adds bloat to the code to declare an async function and then call it. One of our most important rules around snippets in docs is simplicity: remove any code that is not absolutely necessary

Reply:
Thanks for the explanation, sami! i have updated the code.